yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -548,8 +606,19 @@ private[spark] class Client(
     logInfo("Setting up container launch context for our AM")
     val appId = newAppResponse.getApplicationId
     val appStagingDir = getAppStagingDir(appId)
-    val localResources = prepareLocalResources(appStagingDir)
-    val launchEnv = setupLaunchEnv(appStagingDir)
+    val pySparkArchives =
+      if (sys.props.getOrElse("spark.yarn.isPython", null) == "true") {
```

Comment:
could this be `sys.props.getOrElse("...", "false").toBoolean` so it handles cases as well?